### PR TITLE
[Merged by Bors] - Use zap as logger for nipost builder

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation
Use zap logger directly in `nipostBuilder` instead of using our wrapper.

## Changes
Replace calls to `go-spacemesh/log` with calls to `go.uber.org/zap`

## Test Plan
n/a

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

File: activation/nipost.go [hunk 1]
```diff
@@ -317,7 +318,10 @@ func (nb *NIPostBuilder) submitPoetChallenge(
 	if err != nil {
 		return nil, &PoetSvcUnstableError{msg: "failed to get PoET service ID", source: err}
 	}
-	logger := nb.log.WithContext(ctx).WithFields(log.String("poet_id", hex.EncodeToString(poetServiceID.ServiceID)))
+	logger := nb.log.With(
+		log.ZContext(ctx),
+		zap.String("poet_id", hex.EncodeToString(poetServiceID.ServiceID)),
+	)
```
Comment (poszu) [inline activation/nipost.go]: How about logging the URL instead?
- more human-friendly - easier to distinguish poets in logs
- avoids eager encoding in hex

```suggestion
	logger := nb.log.With(log.ZContext(ctx), zap.String("poet", client.Address()))
```

**Review by poszu — APPROVED** [review]

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**